velox/functions/sparksql/Map.cpp

```diff
@@ -96,21 +98,27 @@ class MapFunction : public exec::VectorFunction {
     auto offsets = mapResult->mutableOffsets(rows.end());
     auto rawOffsets = offsets->asMutable<int32_t>();
 
+    // Setting keys and value elements
+    auto keysResult = mapResult->mapKeys();
+    auto valuesResult = mapResult->mapValues();
+    auto baseOffset = keysResult->size();
+    VELOX_CHECK(
```

Comment:
This check is not correct. It is allows for keys and maps vectors to have different sizes. You need to use max of the sizes as baseOffset.

Reply:
Updated, but I'm curious in which case the keys size doesn't equal values size, as we use same `rawOffsets_`and `rawSizes_` to access both keys and values.